docs/design/query-caching.md

```diff
@@ -0,0 +1,224 @@
+The cache will store many representations of database data due to the variations of SQL queries generated and executed against the database. Some queries may include or exclude certain database columns depending on the role of the requestor. 
+
+__Authorization configuration does not explicitly affect how the caching service operates__ because all authorization actions execute prior cache interaction. Requests will be rejected due to authorization restrictions prior to cache interaction.
+
+__DAB will return unexpired cache entries when a database goes offline.__ Depending on the developer configured ttl for an entity, the cache entries will persist in an offline database scenario for the ttl defined duration. A cache entry is evicted after the ttl is surpassed.
+
+### Data in transit
+
+__Data returned by the database is received over the same TLS connection established to execute queries.__ The caching service described in this document is an in-memory cache and not a distributed cache. Consequently, the cache does not further transmit data returned by the database, by design. 
+
+### Access token metadata usage
+
+__Cache entries will not be created for database query text generated from database policies.__ This is because database policies directly affect values set within SQL Server's `-set-session-context` variables and is an implementation detail of database specific query executors. The caching service will be implemented independent of and before execution of query executor code. 
+
+Access tokens directly influence how database policies are resolved. Access tokens are also short lived and token contents used to generate a SQL query may change from token to token. This adds an additional burden to the developer to remember to configure their cache entry `ttl` to properly align with access token expiration. We can't know how frequently token contents may change. Cache entries created from token contents are specific to individual users. We should aim to prevent the cache from favoring specific users because we can then avoid have individual users flood the cache for themselves. 
+
+## Configuration Changes
+
+Entity object have the following top-level properties: source, graphql, rest, permissions, relationships, mappings. Caching will introduce a new top-level property for entities to enable entity-level granularity in cache settings.
+
+```json
+{
+  "entities": {
+    "EntityA": {
+      "cache": {
+        "enabled": true,
+        "ttl": 123
+      }
+    }
+  }
+}
+```
+
+| Property | JSON Data Type | Default Value | Description Value | Required |
+|----------|------------|-------------|---------------|----------|
+| enabled  | Boolean | false | Whether caching is enabled for read requests on this entity. |  Yes |
+| ttl      | Number | 30 | Number of seconds a cache entry is valid before cache eviction. Min: 1s Max: 24hrs/1440s |  Yes |
```

Comment:
how can we make the fact that ttl is in seconds easily interpreted?

Reply:
I would suggest, if necessary, we do this by updating the property name to `ttl-seconds`

Reply:
sounds good to me: `ttl-seconds`